corehq/apps/registration/static/registration/js/user_login_form.js

```diff
@@ -0,0 +1,147 @@
+hqDefine('registration/js/user_login_form', [
+    'jquery',
+    'knockout',
+    'hqwebapp/js/initial_page_data',
+    'hqwebapp/js/assert_properties',
+    'hqwebapp/js/utils/email',
+    'hqwebapp/js/knockout_bindings.ko',
+], function (
+    $,
+    ko,
+    initialPageData,
+    assertProperties,
+    emailUtils
+) {
+    'use strict';
+
+    let loginController = function (options) {
+        assertProperties.assertRequired(options, [
+            'initialUsername',
+            'passwordField',
+            'passwordFormGroup',
+        ]);
+        let self = {};
+
+        self.checkSsoLoginStatusUrl = initialPageData.reverse('check_sso_login_status');
+        self.passwordField = options.passwordField;
+        self.passwordFormGroup = options.passwordFormGroup;
+        self.passwordFormGroup.hide();
+
+        self.authUsername = ko.observable(options.initialUsername);
+        self.authUsername.subscribe(function (newValue) {
+            if (emailUtils.validateEmail(newValue)) {
+                if (self.continueTextPromise) {
+                    self.continueTextPromise.abort();
+                }
+                self.updateContinueText();
+            }
+        });
+
+        self.continueTextPromise = null;
+        self.defaultContinueText = gettext("Continue");
+        self.continueButtonText = ko.observable(self.defaultContinueText);
+        self.showContinueButton = ko.observable(true);
+        self.showContinueSpinner = ko.observable(false);
+
+        self.isContinueDisabled = ko.computed(function () {
+            return !emailUtils.validateEmail(self.authUsername());
+        });
+
+        self.showSignInButton = ko.observable(false);
+
+        /**
+         * This updates the "Continue" Button text with either "Continue"
+         * or "Continue to <IdentityProvider>".
+         * @param {boolean} expandPasswordField - (optional) if this is true, auto expand password field
+         */
+        self.updateContinueText = function (expandPasswordField = false) {
+            self.continueTextPromise = $.post(self.checkSsoLoginStatusUrl, {
+                username: self.authUsername(),
+            }, function (data) {
```

Comment:
out of curiosity was there a reason this was implemented as a post instead of passing the context to the initial template?

Reply:
yes, we don't want to present the entire list of supported IdP's and email domains to be read in javascript for security/privacy purposes. This was discussed outside of this thread.